tests/periph/freqm/main.c

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (C) 2023 ML!PA Consulting GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     tests
+ * @{
+ *
+ * @file
+ * @brief       Application to test functionality of the frequency meter
+ *              peripheral
+ *
+ * @author      Urs Gompper <urs.gompper@ml-pa.com>
+ * @}
+ */
+
+#include <stddef.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <stdint.h>
+
+#include "periph/freqm.h"
+
+volatile bool freqm_buffer_ready = false;
+extern struct freqm_config config_freqm;
+
+void freqm_event_handler(void *context)
+{
+    (void)(context);
+    freqm_buffer_ready = true;
+}
+
+int main(void)
+{
+    uint32_t measuredclock_freq;
+    int block = FREQM_BLOCKING;
+
+    puts("FREQM peripheral driver test");
+
+    /* Initialize frequency meter peripheral */
+    if (freqm_init(&config_freqm) != 0) {
+        return EXIT_FAILURE;
+    }
+
+    /* Register callback handler for FREQM interrupt */
+    freqm_callback_register(freqm_event_handler, NULL);
+
+    /* Start frequency measurement */
+    if (freqm_start(block) == false) {
+        return EXIT_FAILURE;
+    }
+
+    /* Wait for measurement to be finished*/
+    if (!block) {
+        while (freqm_buffer_ready != true) {}
+    }
+
+    /* Read measuremant */
+    if (!freqm_frequency_get(&config_freqm, &measuredclock_freq)) {
+        printf("Measured Clock Frequency: %ld Hz\n", measuredclock_freq);
+    }
+    else {
+        return EXIT_FAILURE;
+    }
+
+    /* main thread exits */
+    return 0;
```

Comment:
You could add another print statement to tell users that the test is over. First time I ran the test, I looked at my terminal expecting more output to come.